crates/subspace-gateway/src/commands/http/server.rs

```diff
@@ -0,0 +1,151 @@
+use actix_web::{web, App, HttpResponse, HttpServer, Responder};
+use serde::{Deserialize, Deserializer, Serialize};
+use std::default::Default;
+use std::sync::Arc;
+use subspace_core_primitives::hashes::{blake3_hash, Blake3Hash};
+use subspace_core_primitives::pieces::PieceIndex;
+use subspace_core_primitives::BlockNumber;
+use subspace_data_retrieval::object_fetcher::ObjectFetcher;
+use subspace_data_retrieval::piece_getter::PieceGetter;
+use tracing::{debug, error, trace};
+
+pub(crate) struct ServerParameters<PG>
+where
+    PG: PieceGetter + Send + Sync + 'static,
+{
+    pub(crate) object_fetcher: ObjectFetcher<PG>,
+    pub(crate) indexer_endpoint: String,
+    pub(crate) http_endpoint: String,
+}
+
+#[derive(Serialize, Deserialize, Debug, Default)]
+#[serde(rename_all = "camelCase")]
+struct ObjectMapping {
+    hash: Blake3Hash,
+    piece_index: PieceIndex,
+    piece_offset: u32,
+    #[serde(deserialize_with = "string_to_u32")]
+    block_number: BlockNumber,
```

Comment:
This looks like a bug to me. Why would you want to receive a number as a string in JSON instead of an actual number? This should be a number, just like al the others in this very struct.

Reply:
Yeah I thought that was a bit weird! The block number provided by the node in the mapping subscription is a number:
https://github.com/autonomys/subspace/blob/3d5483e2b6f2fbe5cd7b7ca613f527164daee611/crates/sc-consensus-subspace-rpc/src/lib.rs#L230-L237

(Specifically a `u32`.)